hal/ia32/pci.c

```diff
@@ -122,21 +108,19 @@ int hal_pciSetBusmaster(pci_dev_t *dev, u8 enable)
 int hal_pciGetDevice(pci_id_t *id, pci_dev_t *dev, void *caps)
 {
 	spinlock_ctx_t sc;
-	unsigned char b, d, f, i;
+	unsigned int b, d, f, i;
 	u32 dv, cl, mask, val;
 	int res = EOK;
 
 	if ((id == NULL) || (dev == NULL))
 		return -EINVAL;
 
+	hal_spinlockSet(&pci_common.spinlock, &sc);
```

Comment:
for 0...255 (for 0 to 32 (for 0 to 8))) That's quite some time to keep spinlock on. Do we need to keep spinlock this long?

Reply:
I changed setting the spinlock for whole PCI bus enumeration for better readability (I figured it's not that much time). If you think it's too much I can restore setting the spinlock for single PCI configuration space access.

Reply:
Maybe hal_spinlockSet() set before:
```
for (d = dev->dev; d < 32; d++) 
```
or before
```
for (f = dev->func; f < 8; f++)
```